races.py

```diff
@@ -0,0 +1,125 @@
+#I need import this because I have s problem with encoding in my notebook
+
+
+page_list = list(range(0, 91))
+
+for p in page_list:
+    #filename = "races.csv"
+    #f = open(filename, "w")
+
+    page = str(p)
+
+    my_url = 'https://ceskybeh.cz/terminovka/?region=0&dfrom=01.%2001.%202019&dto=31.%2003.%202021&rlength=0&rtype=0&advanced=1&search=&page=' + page
+
+    uClient = uReq(my_url)
+    page_html = uClient.read()
+    uClient.close()
+    page_soup = soup(page_html, "html.parser")
+
+
+    # This function find list with 35 items, which have selector/locator "class = row".
+    # Unfortunately I am not able to find more relevant selector in the web page to find a specific container with races.
+    # print(len(containers))
+    containers = page_soup.findAll("div", { "class" : "row"})
+
+    # I found by trial-error method that container with relevant content is the 12th one.
+    # So I need to select container with index 11.
+    # print(container)
+    container = containers[11]
+
+    # This function will find a list of items, which contains two kinds of span with "class = text-muted iframe-hidden".
+    # First is for date of the race, the second for the place of the race.
+    date_place = container.findAll("span", {"class" : "text-muted iframe-hidden"})
+    # I decided to take a place of the race from another (for me more suitable part) of the webpage, so right now
+    # I need only each second item from the list (starting from the first item).
+    # I decided delete every second item from the list (starting from the second item).
+    del date_place[1::2]
+
+    dates = date_place
+
+    for d in dates:
+        d = dates[0]
+        date_with_days = d.text
+        sliced_date = date_with_days.split(' ')
+
+        day_with_one_digit = int(sliced_date[1].strip('.'))
+        if day_with_one_digit < 10:
+            day = '0' + str(day_with_one_digit)
+        else:
+            day = str(day_with_one_digit)
+
+        #Because of our database I need to change names of months to numbers, 
+        #starting with 0, if the number of the month is smaller than 10. At first I made it with a condition,
+        #but it is a quite long and same code. So I tried to make it through dictionary.
+        #Keys of dictionary are names of months in the form of the word the same as at the page.
+        names_of_months = ['ledna', 'února', 'března', 'dubna', 'května', 'června', 'července', 'srpna', 'září', 'října', 'listopadu', 'prosince']
+        #print(names_of_months)
+        #values are numbers for months, for example january = 1, etc.
+        values_of_months = list(range(1, 13))
+        #print( values_of_months)
+        months = dict(zip(names_of_months,  values_of_months))
+        #print(months)
+        month = sliced_date[2]
+        if month in months:
+            month = '0' + str(months[month])
+            #print(month)
+        else:
+            print("U závodu není uvedeno datum v potřebném formátu.")
+
```

Comment:
I'd propose either terminating the whole problem with an exception here when you fail to parse the date because otherwise you'd end up with a race that you have no date for, so you wouldn't be able to insert it into the database, for instance.

Generally, I'd say that you want to abort the program (raise some exception) since not being able to parse a date means your program needs to be fixed to handle all the data coming from the website (rather than waiting/asking the website to fix their stuff).

Reply:
Hope I chose the right version of Exception?

Reply:
I'd propose using `RuntimeError` instead of `Exception` - `Exception` is a general catch-all kind of exception, while `RuntimeError` is a more specific exception, describing that a closer unspecified error has occurred during runtime.